### PR TITLE
add cleanup components

### PR DESCRIPTION
**Pull request by tstringer**
I recently created [`boiler-room-custodian`](https://www.npmjs.com/package/boiler-room-custodian) which is designed to take awesome boilerplates (like yours that you have here), and yank out the unnecessary sample code that is often included to show a working example.

This pull request implements `boiler-room-custodian`, and if you run `npm run cleanup` it'll remove the sample code files, unnecessary lines, etc. so that the developer can start custom development right away.

This was implemented and merged into the [electron-react-boilerplate repository (link: PR)](https://github.com/chentsulin/electron-react-boilerplate/pull/402).

Let me know your thoughts on this!

### CODE REVIEW

File: setup.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module.exports = {
+    remove: [
+        { file: 'resources/icons/512x512.png' },
+        { file: 'resources/osx/dmg-background.png' },
+        { file: 'resources/osx/dmg-background@2x.png' },
+        { file: 'resources/osx/dmg-icon.icns' },
+        { file: 'resources/osx/icon.icns' },
+        { file: 'resources/windows/icon.ico' },
+        { file: 'resources/windows/setup-icon.ico' },
+        { folder: 'src/hello_world' }
```
Comment (blainesch) [inline setup.js]: `setup.js`

Comment (tstringer) [inline setup.js]: Not sure I understand..?  I don't see any suggested "changes", let me know if I'm missing something! 😄

Comment (blainesch) [inline setup.js]: I was suggesting you should add `setup.js` to the list of files to delete, since you'll never need it again.

Comment (tstringer) [inline setup.js]: Ohhhh that's a good point!  Great idea, and totally agreed.  Perhaps I should bake that into the product... Either way, for now I'll add it explicitly.  Thanks! 👍  🚀

**Comment by tstringer** [comment]
`setup.js` added to the cleanup script! 😄

**Comment by szwacz** [comment]
Hi, this is interesting idea, but I have a lot of "but".

General problems
- Solution like that should be completely self-removing after use. What means also dependency and script from `package.json` should be sweeped out.
- This feature is dangerous when someone will not use it at the beginning. Then after few months of development someone can spot it again and "hey, I wonder what this thing does" remove some vital code. Assuming he uses git `git checkout .` will cleanup the mess, but still this feature ages badly.

Problems for this specific boilerplate
- There is really not much code to be removed: `hello world` directory and few lines in `app.js` and `app.html`.
- `README.md` really shuldn't be cleaned up, because most of informations there are just about how to use the project (exactly what readmes are for). 
- `resources` folder shouldn't be touched by your tool as well. First of all icons provided by default are perfectly fine for student projects or what have you. And in professional uses names and sizes of those icons do matter, so it's easier to replace them one by one, than dig through original boilkerplate repo to recall "what extension and name this file should have".

Your tool is really great for more complicated boilerplates, but the tradeoff of added complexity to gained convenience for this particular boilerplate doesn't look convincing so far.

**Comment by black-snow** [comment]
Why would you delete stuff that you need for the release anyway?

**Comment by tstringer** [comment]
@szwacz makes sense, thanks for the consideration!

**Pull request closed without merging**